src/test/java/de/tum/in/www1/artemis/exercise/ExerciseUtilService.java

```diff
@@ -113,24 +112,25 @@ public Exercise addMaxScoreAndBonusPointsToExercise(Exercise exercise) {
      * @param exercise The exercise to which the grading instructions should be added.
      * @return The list of grading criteria.
      */
-    public List<GradingCriterion> addGradingInstructionsToExercise(Exercise exercise) {
+    public Set<GradingCriterion> addGradingInstructionsToExercise(Exercise exercise) {
         GradingCriterion emptyCriterion = ExerciseFactory.generateGradingCriterion(null);
         List<GradingInstruction> instructionWithNoCriteria = ExerciseFactory.generateGradingInstructions(emptyCriterion, 1, 0);
```

Comment:
This method should also return a Set.

Reply:
IIRC there were a few cases where the caller then further adapted each item in the list with some additional changes. This is of course easier when it remains a list where you can directly index items. I’ll check again, though.

Reply:
Apparently not any more after the other refactorings. Changed to a set in e430ce60b99b847a13bb5dcba691b5e2bf960078.